src/libraries/System.Diagnostics.DiagnosticSource/src/System/Diagnostics/Activity.cs

```diff
@@ -686,8 +687,7 @@ public void Stop()
                 }
 
                 Source.NotifyActivityStop(this);
-
-                SetCurrent(Parent);
+                SetCurrent(Parent ?? ((_originalParent != null && !_originalParent.IsFinished) ? _originalParent : null));
```

Comment:
Parent is either _originalParent or null. Combining the previous comment + this one should reduce the implementation to:
````C#
SetCurrent(_originalParent);
````